core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -2384,27 +2386,79 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
     * }}}
     * @throws scala.IllegalArgumentException if `n` <= 0
     */
-  def sliding(n: Int): Stream[F, collection.immutable.Queue[O]] = {
-    require(n > 0, "n must be > 0")
-    def go(
-        window: collection.immutable.Queue[O],
-        s: Stream[F, O]
-    ): Pull[F, collection.immutable.Queue[O], Unit] =
-      s.pull.uncons.flatMap {
-        case None => Pull.done
-        case Some((hd, tl)) =>
-          val (out, carry) = hd.scanLeftCarry(window)((w, i) => w.dequeue._2.enqueue(i))
-          Pull.output(out) >> go(carry, tl)
-      }
-    this.pull
-      .unconsN(n, true)
-      .flatMap {
-        case None => Pull.done
-        case Some((hd, tl)) =>
-          val window = hd.foldLeft(collection.immutable.Queue.empty[O])(_.enqueue(_))
-          Pull.output1(window) >> go(window, tl)
-      }
-      .stream
+  def sliding(n: Int): Stream[F, collection.immutable.Queue[O]] = sliding(n, 1)
```

Comment:
Hm, should this return `Chunk.Queue` instead of `c.i.Queue`? Performance should be better in general that way...

Reply:
Yep, I'll change it and measure the perfomance difference.

But, this how it won't be backward compatible.